core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -143,6 +143,10 @@ private TableProperties() {
   public static final String DELETE_ORC_BLOCK_SIZE_BYTES = "write.delete.orc.block-size-bytes";
   public static final long ORC_BLOCK_SIZE_BYTES_DEFAULT = 256L * 1024 * 1024; // 256 MB
 
+  public static final String ORC_VECTOR_ROW_BATCH_SIZE = "write.orc.vectorized-row-batch-size";
```

Comment:
We are replacing the old `iceberg.orc.vectorbatch.size` with the newly introduced `writer.orc.vectorized-row-batch-size` here , right ?   In this way,  the old table using the old config key won't take any effect in the next release.  I think we'd better to keep the original approach.

Reply:
`iceberg.orc.vectorbatch.size` doesn't fit our naming conventions. Previously, it was a private variable and was not exposed as a public table configuration. In my humble opinion, if we formally expose it as a table configuration,
, would it be better to use a standard name?

> Config naming
> 1.Use - to link words in one concept
  >     For example, preferred convection access-key-id rather than access.key.id
> 2.Use . to create a hierarchy of config groups
  >     For example, s3 in s3.access-key-id, s3.secret-access-key

Reply:
It's true that the original name `iceberg.orc.vectorbatch.size` is not an iceberg name style. This config key was introduced in this PR: https://github.com/apache/iceberg/pull/139/files.  In theory, people could add this config key-value in their enviroment  hadoop configuration file.  But after we merged this PR, the key won't take any effect even if we use the old key in hadoop environment. 

I will suggest to make it deprecated and then introduce the correct table property. But I think we still need to be able to read the value from the old key.

Reply:
It's true that the original name `iceberg.orc.vectorbatch.size` is not an iceberg name style. This config key was introduced in this PR: https://github.com/apache/iceberg/pull/139/files.  In theory, people could add this config key-value in their enviroment  hadoop configuration file.  But after we merged this PR, the key won't take any effect even if we use the old key in hadoop environment. 

I will suggest to make it deprecated and then introduce the correct table property. But I think we still need to be able to read the value from the old key.

Reply:
Okay, I'll restore it, and let's do it separately later if we think it's necessary.

Reply:
OK, I added the standardized table configuration and tried to get values from the old key for compatibility, please take a look, thx.

Reply:
I agree that we should deprecate and keep supporting `iceberg.orc.vectorbatch.size` from the Hadoop conf since that slipped through review. But we can still clean this up with a reasonable table property.

The read property is `read.orc.vectorization.batch-size` so I think this should match: `write.orc.vectorization.batch-size`